### PR TITLE
Api Review Changes

### PR DESCRIPTION
**Pull request by ibrandes**
- making getEncryptionScope protected and editing cpkn async test
- chaning isRenameSupported
- removing bool readtofile overload
- adding readtofile and getproperties methods to async clients
- changing upn to UserPrincipalNam
-  making sync and async getPropertiesWithResponse upn methods private, and fixing the overloads that were breaking changes

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java [hunk 1]
```diff
@@ -258,7 +258,7 @@ protected BlobAsyncClientBase(HttpPipeline pipeline, String url, BlobServiceVers
      *
      * @return the encryption scope used for encryption.
      */
-    public String getEncryptionScope() {
+    protected String getEncryptionScope() {
```
Comment (ibrahimrabab) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java:261]: Why are we changing this back to protected? Why was this changed to public in the first place?

Comment (ibrandes) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java:261]: It was changed to public in the first place for testing purposes and to be more consistent with the sync client as the sync client has this method as public. However, in the API review, this change was determined to be breaking so it is being switched back to protected and the test (getEncryptionScopeClient() in CPKNAsyncTests.java) has been rewritten.

**Review by ibrahimrabab — APPROVED** [review]

**Review by alzimmermsft — CHANGES_REQUESTED** [review]

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakePathAsyncClient.java [hunk 2]
```diff
@@ -902,6 +929,37 @@ public Mono<Response<PathProperties>> getPropertiesWithResponse(DataLakeRequestC
...
+     * <!-- end com.azure.storage.file.datalake.DataLakePathAsyncClient.getPropertiesWithResponse#PathGetPropertiesOptions -->
+     *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/rest/api/storageservices/get-blob-properties">Azure Docs</a></p>
+     *
+     * @param options {@link PathGetPropertiesOptions}
+     * @return A reactive response containing the resource's properties and metadata.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<PathProperties>> getPropertiesWithResponse(PathGetPropertiesOptions options) {
```
[the remark this answers is not in the document]
Comment (ibrandes) [reply sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakePathAsyncClient.java]: Decided to remove this and the sync method and make it private.
